### PR TITLE
Add configuration of useOutputWalkers for doctrine paginator

### PR DESCRIPTION
**Pull request by rn0**
Usage:

```php
         return $factory->createDataSource(
             'doctrine-orm',
             ['qb' => $repository->createQueryBuilder('c'), 'useOutputWalkers' => false],
              $this->getId()
          )->setMaxResults(30);
```

### CODE REVIEW

**Comment by JarekW** [comment]
can you describe what was the problem?

File: lib/Driver/Doctrine/ORM/DoctrineDriver.php [hunk 1]
```diff
@@ -50,15 +50,21 @@ class DoctrineDriver extends DriverAbstract
      * @var \Doctrine\ORM\QueryBuilder
      */
     private $currentQuery;
+    /**
```
Comment (chives) [inline lib/Driver/Doctrine/ORM/DoctrineDriver.php]: missing empty line

**Comment by rn0** [comment]
There is this code in `Doctrine\ORM\Tools\Pagination\Pagiantor`

```
            if ($this->useOutputWalker($subQuery)) {
                $subQuery->setHint(Query::HINT_CUSTOM_OUTPUT_WALKER, 'Doctrine\ORM\Tools\Pagination\LimitSubqueryOutputWalker');
            } else {
                $this->appendTreeWalker($subQuery, 'Doctrine\ORM\Tools\Pagination\LimitSubqueryWalker');
            }
```

```
    private function useOutputWalker(Query $query)
    {
        if ($this->useOutputWalkers === null) {
            return (Boolean) $query->getHint(Query::HINT_CUSTOM_OUTPUT_WALKER) == false;
        }

        return $this->useOutputWalkers;
    }
```

Therefore, query uses `LimitSubqueryOutputWalker` as output watcher. I have query with sorting for entity with class table inheritance for which default `LimitSubqueryOutputWalker` do not work (it produces invalid sql). Workaround for this problem is to use `LimitSubqueryWalker` which produces valid sql for my case.

**Comment by rn0** [comment]
@fsi-open

**Comment by chives** [comment]
this PR should go only to current master (1.3-dev) branch as long as it's a new feature

**Pull request merged**